addons/statemachine/example.hpp

```diff
@@ -0,0 +1,32 @@
+/*
+    This is an example for a CBA state machine config that can be read by the
+    CBA_statemachine_fnc_createFromConfig function.
+    This example would result in the same state machine as the one from
+    example.sqf.
+*/
+
+class MyAddon_Statemachine {
+    // Class properties have the same name as the corresponding function parameters
+    // and code goes into strings.
+    list = "allGroups select {!isPlayer leader _x}";
+
+    // States are just subclasses of the state machine
+    class Initial {
+        onState = "";
+        onStateEntered = "";
+        onStateLeaving = "";
```

Comment:
That's on purpose, it follows an event naming convention from Microsoft. The point is that it tells you already exactly when this happens:
- entered means you're already within the new state (thus past-tense)
- leaving means you're still in the state, but are about in progress of leaving it (thus gerund)

More info: https://msdn.microsoft.com/de-de/library/h0eyck3s(v=vs.71).aspx